flow/SimpleOpt.h

```diff
@@ -882,18 +900,18 @@ int CSimpleOptTempl<SOCHAR>::CalcMatch(const SOCHAR* a_pszSource, const SOCHAR*
 
 	// determine the argument type
 	int nArgType = SO_O_ICASE_LONG;
-	if (a_pszSource[0] != '-') {
+	if (a_pszSource[0] != '_') {
```

Comment:
My only concern with this approach is that we're also changing the leading hyphens, such that it would be legal to call this with `__cluster_file`. I'm thinking we should avoid converting a single or double leading hyphen and only convert any other hyphens present.

Reply:
That makes sense and easy to be implemented I think. For this issue, I consider there are two available solutions:
1. As your suggestion, instead of transferring all the hyphens to underscores for the internal flags and user's input, we could change the internal flags to the form which has leading hyphens but all remaining hyphens are converted to underscore. 
    And for the user's input flags, we skip the leading hyphens (if any), then transfer the remaining hyphens to underscores to do the match.

2. Or maybe we can simply check if the user's input is legal: if there is leading underscore then it's invalid.

I think the second one is easier. Which one do you think is better?